CosmicsCalib/src/CaloCosmicEnergy_module.cc

```diff
@@ -0,0 +1,746 @@
+    _diagLevel(pset.get<int>("diagLevel", 0)),
+    CutNCryHit(pset.get<int>("CutNCryHit", 3)),
+    CutEnergyDep(pset.get<float>("CutEnergyDep", 10.)),
+    CutChi2Norm(pset.get<float>("CutChi2Norm", 2.5)),
+    _caloClusterTag(pset.get<art::InputTag>("caloClusterCollection")),
+    _caloHitTag(pset.get<art::InputTag>("caloHitCollection")){}
+
+  void CaloCosmicEnergy::beginJob(){
+    if (_diagLevel > 0 ) std::cout<< "CaloCosmicEnergy: Entering beginJob" << std::endl;
+ 
+    for(int iCry=0; iCry < nDisks*nCrystals; iCry++){
+      for(int iSiPM=0; iSiPM < nSiPMs; iSiPM++){
+	hSiPM[iCry][iSiPM] = tfdir.make<TH1F>(Form("Crystal_%i_SiPM_%i",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d",iSiPM, iCry),70, 15., 50.);
+	hSiPMv[iCry][iSiPM] = tfdirv.make<TH1F>(Form("Crystal_%i_SiPM_%i_v",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d only vertical tracks",iSiPM, iCry),70, 15., 50.);
+	hSiPMd[iCry][iSiPM] = tfdird.make<TH1F>(Form("Crystal_%i_SiPM_%i_d",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d only diagonal tracks",iSiPM, iCry),70, 15., 50.);
+	hSiPMfp[iCry][iSiPM] = tfdirfp.make<TH1F>(Form("Crystal_%i_SiPM_%i_fp",iCry, iSiPM),Form("Energy deposited in SiPM %i crystal %d normalized track length",iSiPM, iCry),70, 15., 50.);
+      }
+      for(int ibin=0; ibin < Ebin; ibin++){
+	CryALR[iCry][ibin] = new TH1F(Form("Asymmetry_Crystal_%i_band_%i", iCry, ibin), Form("Asymmetry of crystal %i in band [%i, %i)",iCry, (ibin+1)*5, (ibin+1)*5+5), 200,-1., 1.);
+      }
+    }
+    MPV = tfdir_res.make<TH1F>("MPV_distr","MPV distribution",400, 10., 30.);
+    MPVv = tfdir_res.make<TH1F>("MPV_v_distr","MPV distribution only vertical tracks",400, 10., 30.);
+    MPVd = tfdir_res.make<TH1F>("MPV_d_distr","MPV distribution only diagonal tracks",400, 10., 30.);
+    MPVfp = tfdir_res.make<TH1F>("MPV_fp_distr","MPV distribution all normalized tracks",400, 10., 30.);
+    ALR_all = tfdir_alr.make<TH2F>("ALR","Asymmetry left right all energies",600,0.,600., 100, -1., 1.);
+    LR_all = tfdir_alr.make<TH1F>("ALR_All_Energies", "Asymmetry Left Right all energies", 200, -1, 1.);
+    for(int i =0; i<Ebin; i++){
+      LR[i] = tfdir_alr.make<TH1F>(Form("LR_ene_band_%i",i), Form("Ratio Left Right Energy band [%i - %i) MeV", i*5, (i+1)*5), 100, 0.5, 1.5);
+      ALR[i] = tfdir_alr.make<TH1F>(Form("ALR_ene_band_%i",i), Form("Asymmetry Left-Right Energy band [%i - %i) MeV", i*5, (i+1)*5), 200, -1., 1.);
+    }
+  }//end begin job
+
+
+  void CaloCosmicEnergy::analyze(art::Event const& event){
+    art::ServiceHandle<GeometryService> geom;
+    if ( !geom->hasElement<Calorimeter>() ){
+      throw cet::exception("NO-CALO-GEOM") 
+	<< "CaloEnergy: Calorimeter geometry not found";
+    }
+
+    const Calorimeter& cal = *(GeomHandle<Calorimeter>());
+
+    art::Handle<CaloClusterCollection> caloClustersHandle;
+    event.getByLabel(_caloClusterTag, caloClustersHandle);
+    const CaloClusterCollection& caloClusters(*caloClustersHandle);
```

Comment:
THe preferred idiom for getting a data product is show in the Tutorials at the line:

https://github.com/Mu2e/Tutorial/blob/9db1568f23e83167a97ec3af110170047441e3ea/AllInOne/src/StrawDigiStudy_module.cc#L95

You need to work back from that to see the definition of the token and the tag.